src/main/java/org/jvnet/hudson/test/TestPluginManager.java

```diff
@@ -1,175 +1,56 @@
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
-
 package org.jvnet.hudson.test;
 
-import hudson.LocalPluginManager;
+import java.io.IOException;
+
 import hudson.Plugin;
 import hudson.PluginManager;
 import hudson.PluginWrapper;
 import hudson.Util;
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.lang.reflect.Method;
-import java.net.URISyntaxException;
-import java.net.URL;
-import java.nio.charset.StandardCharsets;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.Set;
-import java.util.jar.Manifest;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-import org.apache.commons.io.FileUtils;
-import org.junit.Assert;
 
-/**
+/** 
  * {@link PluginManager} to speed up unit tests.
- *
- * <p>
- * Instead of loading every plugin for every test case, this allows them to reuse a single plugin manager.
- *
- * <p>
- * TODO: {@link Plugin} start/stop/postInitialize invocation semantics gets different. Perhaps
- * 
- * @author Kohsuke Kawaguchi
- * @see HudsonTestCase#useLocalPluginManager
- */
-public class TestPluginManager extends PluginManager {
-    public static final PluginManager INSTANCE;
+* 
+*
+* <p>
+* Instead of loading every plugin for every test case, this allows them to reuse a single plugin manager.
+* This has the downside that you can not disabling plugins.
+* <p>
+* TODO: {@link Plugin} start/stop/postInitialize invocation semantics gets different. Perhaps
+* 
+* @author Kohsuke Kawaguchi
+* @see HudsonTestCase#useLocalPluginManager
+*/
+public class TestPluginManager extends UnitTestSupportingPluginManager {
 
-    public TestPluginManager() throws IOException {
-        // TestPluginManager outlives a Jetty server, so can't pass in ServletContext.
-        super(null, Util.createTempDir());
-    }
+    public static final PluginManager INSTANCE;
 
-    /** @see LocalPluginManager#loadBundledPlugins */
-    @Override
-    protected Collection<String> loadBundledPlugins() throws Exception {
```

Comment:
Unfortunately Git provides no way to rename a file and then create a distinct file with the original name in such a way as to make for a reviewable diff. So I have no idea what this PR is doing, and unfortunately we cannot rename `TestPluginManager` since there are a few external references to it.

Reply:
see #215 which has a separate commit for the rename and code changes.